components/script/dom/htmlinputelement.rs

```diff
@@ -875,6 +875,18 @@ impl HTMLInputElement {
                 content.strip_newlines();
                 content.strip_leading_and_trailing_ascii_whitespace();
             }
+            atom!("date") => {
+                let mut textinput = self.textinput.borrow_mut();
+                if !textinput.single_line_content_mut().is_valid_date_string() {
+                    textinput.set_content("".into());
+                }
+            }
+            atom!("month") => {
+                let mut textinput = self.textinput.borrow_mut();
+                if ! textinput.single_line_content_mut().is_valid_month_string() {
```

Comment:
@KiChjang
even I set
```
            atom!("month") => {
                let mut textinput = self.textinput.borrow_mut();
                textinput.set_content("".into());
            }
```
It will still return the origin value, like
```
  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/month.html:
  │ FAIL [expected PASS] Month should be > 0: If the value of the element is not a valid month string, then set it to the empty string instead.>
  │   → assert_equals: expected "" but got "2013-00"
  │ 
  │ @http://web-platform.test:8000/html/semantics/forms/the-input-element/month.html:64:9
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1489:20
  │ test@http://web-platform.test:8000/resources/testharness.js:511:9
  └ @http://web-platform.test:8000/html/semantics/forms/the-input-element/month.html:63:7

  ▶ Unexpected subtest result in /html/semantics/forms/the-input-element/month.html:
  │ FAIL [expected PASS] Month should be <= 13: If the value of the element is not a valid month string, then set it to the empty string instead.
  │   → assert_equals: expected "" but got "2013-13"
  │ 
  │ @http://web-platform.test:8000/html/semantics/forms/the-input-element/month.html:60:9
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1489:20
  │ test@http://web-platform.test:8000/resources/testharness.js:511:9
  └ @http://web-platform.test:8000/html/semantics/forms/the-input-element/month.html:59:7
```

Reply:
I just saw there's a space between `!` and `textinput`.
Not sure if it will be an issue for Rust?

Reply:
So, my guess here is that the `value` content attribute is not properly sanitized when it is first parsed. In other words, `bind_to_tree` needs to call `sanitize_value`.

Reply:
@CYBAI Although this is not proper style, there's no problem with adding a whitespace in-between `!` and the expression that it is trying to negate, see https://play.rust-lang.org/?gist=d27e1e8f115cb01c70bf97b8c5312c60&version=stable.

Reply:
KiChjang  I just follow `color` method, and seems `color` implement is correct (It passes tests). So..?

Reply:
@KiChjang  You are right. Initialization works.
But I am wondering why `color` is fine  lol

Reply:
Many other tests pass when I set initialization!